src/openrct2-ui/windows/ServerList.cpp

```diff
@@ -73,529 +71,519 @@ static Widget window_server_list_widgets[] = {
-            break;
-        case WIDX_LIST:
+        switch (widgetIndex)
         {
-            int32_t serverIndex = w->selected_list_item;
-            if (serverIndex >= 0 && serverIndex < static_cast<int32_t>(_serverList.GetCount()))
+            case WIDX_CLOSE:
+                Close();
+                break;
+            case WIDX_PLAYER_NAME_INPUT:
+                WindowStartTextbox(*this, widgetIndex, STR_STRING, _playerName, 63);
+                break;
+            case WIDX_LIST:
             {
-                const auto& server = _serverList.GetServer(serverIndex);
-                if (server.IsVersionValid())
+                int32_t serverIndex = selected_list_item;
+                if (serverIndex >= 0 && serverIndex < static_cast<int32_t>(_serverList.GetCount()))
                 {
-                    JoinServer(server.Address);
-                }
-                else
-                {
-                    Formatter ft;
-                    ft.Add<const char*>(server.Version.c_str());
-                    ContextShowError(STR_UNABLE_TO_CONNECT_TO_SERVER, STR_MULTIPLAYER_INCORRECT_SOFTWARE_VERSION, ft);
+                    const auto& server = _serverList.GetServer(serverIndex);
+                    if (server.IsVersionValid())
+                    {
+                        JoinServer(server.Address);
+                    }
+                    else
+                    {
+                        Formatter ft;
+                        ft.Add<const char*>(server.Version.c_str());
+                        ContextShowError(STR_UNABLE_TO_CONNECT_TO_SERVER, STR_MULTIPLAYER_INCORRECT_SOFTWARE_VERSION, ft);
+                    }
                 }
+                break;
             }
-            break;
+            case WIDX_FETCH_SERVERS:
+                ServerListFetchServersBegin();
+                break;
+            case WIDX_ADD_SERVER:
+                TextInputOpen(widgetIndex, STR_ADD_SERVER, STR_ENTER_HOSTNAME_OR_IP_ADDRESS, {}, STR_NONE, 0, 128);
+                break;
+            case WIDX_START_SERVER:
+                ContextOpenWindow(WindowClass::ServerStart);
+                break;
         }
-        case WIDX_FETCH_SERVERS:
-            ServerListFetchServersBegin();
-            break;
-        case WIDX_ADD_SERVER:
-            WindowTextInputOpen(w, widgetIndex, STR_ADD_SERVER, STR_ENTER_HOSTNAME_OR_IP_ADDRESS, {}, STR_NONE, 0, 128);
-            break;
-        case WIDX_START_SERVER:
-            ContextOpenWindow(WindowClass::ServerStart);
-            break;
     }
-}
 
-static void WindowServerListResize(WindowBase* w)
-{
-    WindowSetResize(*w, WWIDTH_MIN, WHEIGHT_MIN, WWIDTH_MAX, WHEIGHT_MAX);
-}
+    void OnResize() override
+    {
+        if (width < min_width)
+        {
+            Invalidate();
+            width = min_width;
+        }
+        if (height < min_height)
+        {
+            Invalidate();
+            height = min_height;
+        }
```

Comment:
You're right, this was me being a little gung-ho in matching the other file I was using as reference. I see now that `WindowSetResize` still seems to be called through out the code, just with a reference to `this` instead of WindowBase. Will make this change in my updated draft.